test/unit/Server/Grabber/GrabberTest.spec.ts

```diff
@@ -62,4 +62,42 @@ describe("GrabberTest", function () {
             )
         });
     });
+    describe("Fetch Course In Session", () => {
+        const course: CourseInfo = {
+                subject: "CPSC",
+                course_id: "100",
+                term: Terms.T2019W
+            };
+
+        it("Return correct course URL", () => {
+            
+            expect(Grabber.getCourseInSessionURL(course)).equals('https://courses.students.ubc.ca/cs/courseschedule?sesscd=W&pname=subjectarea&tname=subj-course&course=100&sessyr=2019&dept=CPSC');
+        });
+
+        it("Return Exception when term is not defined", () => {
+            const noTermCourse:CourseInfo = {
+                subject: "CPSC",
+                course_id: "100"
+            }
+            return Grabber.getCourseInSession(noTermCourse).then(() => {
+                () => expect.fail('Should not resolve')
+            }).catch(e => expect(e).instanceOf(HTMLFetchException));
+        });
+
+        it("Return all sections of the course", () => {
+            return Grabber.getCourseInSession(course).then((data) => {
+                expect(data.length).equals(45);
+                expect(data).contains('CPSC 100 101');
```

Comment:
Also try to change it to a data structure instead of string?

Reply:
What kind of data structure you want for the sections? Right now what we grab is just a list of all section number.

Reply:
Anything better than a string